python/cudf/cudf/core/multiindex.py

```diff
@@ -241,6 +241,76 @@ def labels(self):
         )
         return self.codes
 
+    def isin(self, values, level=None):
+        """Return a boolean array where the index values are in values.
+
+        Compute boolean array of whether each index value is found in
+        the passed set of values. The length of the returned boolean
+        array matches the length of the index.
+
+        Parameters
+        ----------
+        values : set, list-like, Index or Multi-Index
+            Sought values.
+        level : str or int, optional
+            Name or position of the index level to use (if the index
+            is a MultiIndex).
+        Returns
+        -------
+        is_contained : cupy array
+            CuPy array of boolean values.
+
+        """
+        from cudf.utils.dtypes import is_list_like
+
+        if level is None:
+            if isinstance(values, cudf.MultiIndex):
+                values_idx = values
+            elif (
+                (
+                    isinstance(
+                        values,
+                        (
+                            cudf.Series,
+                            cudf.Index,
+                            cudf.DataFrame,
+                            column.ColumnBase,
+                        ),
+                    )
+                )
+                or (not is_list_like(values))
+                or (
+                    is_list_like(values)
+                    and len(values) > 0
+                    and not isinstance(values[0], tuple)
+                )
+            ):
+                raise TypeError(
+                    "values need to be a set, list-like tuple squences, Index \
+                        or Multi-Index."
```

Comment:
We're raising this TypeError when `values` is a `cudf.Index`, this is confusing based on the docstring

Reply:
@kkraus14, Clarified both the Error message in which case it errors and doc with `Note`.